addons/stock/models/stock_production_lot.py

```diff
@@ -253,3 +264,13 @@ def _find_delivery_ids_by_lot(self, lot_path=None, delivery_by_lot=None):
                 # delivery_ids = self._find_delivery_ids(moves_by_lot[lot.id])
             delivery_by_lot[lot.id] = list(delivery_ids)
         return delivery_by_lot
+
+    def _search_product_qty(self, operator, value):
+        domain = [('location_id.company_id', '!=', False), ('location_id.usage', 'in', ('internal', 'transit')), ('lot_id', '!=', False)]
+        quants = self.env['stock.quant'].read_group(domain=domain, fields=['quantity:sum'], groupby=['lot_id'])
+
+        ids = []
+        for quant in quants:
+            if OPERATORS[operator](quant['quantity'], value):
+                ids.append(quant['lot_id'][0])
+        return [('id', 'in', ids)]
```

Comment:
Move this to underneath `def _product_qty(self):` so it's easier to match the 2 functions together + to match the [coding convention](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#symbols-and-conventions) 

Also your domain doesn't quite capture the original compute's filter logic (my suggestion should be tested in case I wrote it wrong)

Another also, the variable name "quants/quant" is confusing since we're looking more unique number of lots records with quants, rather than the unique number of quants, so "lots/lot" are a more accurate name.

Final also, if a lot has no quants, then it will be excluded from the search even though its quantity will display as 0 in the form view. So that use case should be added into the search result

```suggestion

    def _search_product_qty(self, operator, value):
        domain = [('lot_id', '!=', False), '|',  ('location_id.usage', '=', 'internal'),  '&', ('location_id.company_id', '!=', False), ('location_id.usage', '=', 'transit')]
        lots = self.env['stock.quant'].read_group(domain=domain, fields=['quantity:sum'], groupby=['lot_id'])

        ids = []
        for lot in lots:
            if OPERATORS[operator](lot['quantity'], value):
                ids.append(lots['lot_id'][0])
        return [('id', 'in', ids)]
```

Reply:
I've update as you suggest but don't really understand the point about the domain, I just took the same as in ``_product_qty()``